### PR TITLE
Changed grid initialisation.

### PR DESCRIPTION
**Pull request by joakim-hove**
This is a followup to: https://github.com/OPM/opm-grid/pull/216/

### CODE REVIEW

File: examples/initCPGrid.hpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        std::array<double, 3> cellsz = {{ param.getDefault<double>("dx", 1.0),
+                                          param.getDefault<double>("dy", 1.0),
+                                          param.getDefault<double>("dz", 1.0) }};
+        grid.createCartesian(dims, cellsz);
+    } else {
+        OPM_THROW(std::runtime_error, "Unknown file format string: " << fileformat);
+    }
+}
+
+#endif
```
Comment (akva2) [inline examples/initCPGrid.hpp]: no templating here. separate compilation unit please (split in hpp/cpp).

**Comment by akva2** [comment]
with requested split you can merge when it's appropriate to do so.

File: examples/initCPGrid.hpp (hunk 1 above)
Comment (joakim-hove) [inline examples/initCPGrid.hpp]: > [...] stick it in opm/upscaling 

As in extend the upscaling library? The reason I did it the way I did was that (I assumed) the example programs could only consist of one compilation unit. Happy to add header and source files in `opm/upscaling`.

Comment (akva2) [inline examples/initCPGrid.hpp]: the upscaling library is for code shared between applications so it fits the bill perfectly.

File: examples/initCPGrid.hpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
+/*
+  This is small utility to inclded in the example/test programs in opm-upscaling.
```
Comment (akva2) [inline examples/initCPGrid.hpp]: oh and forgot. license header and fix the type inclUded.

**Comment by joakim-hove** [comment]
OK - amend and force push performed.

File: examples/initCPGrid.hpp [hunk 3]
```diff
@@ -0,0 +1 @@
+in
```
Comment (akva2) [inline examples/initCPGrid.hpp]: hmm..

**Comment by joakim-hove** [comment]
> OK - amend and force push performed.

again ....

**Comment by akva2** [comment]
all good. i don't get the love for all those spaces but nothing i care strongly about.

**Comment by joakim-hove** [comment]
A failing test - not ready for merging.

**Comment by akva2** [comment]
'Deck requires a 'GRID' section'. sounds like your changes to me, can you inspect benchmark_tiny_grid.grdecl ? grdecls are mostly greek to me..

**Comment by atgeirr** [comment]
Has this reintroduced the need for a complete deck (again)? That was one reason we could not go all out the EclipseState route before, I assumed that was no longer the case?

**Comment by pgdr** [comment]
@atgeirr No, it's a one-line fix in `setupGridAndProps.hpp#165`.

In `setupGridAndPropsEclipse`, the change is to make the eclipseGrid directly instead of going via state.  I could make a PR to Joakim, but it's probably easier if he makes the change.

**Comment by pgdr** [comment]
@joakim-hove Here's a patch.  In file /opm/porsol/common/setupGridAndProps.hpp

    @@ -165,3 +165,3 @@ namespace Opm
    -  std::shared_ptr<EclipseState> state = std::make_shared<EclipseState>(deck , parseContext);
    +  std::shared_ptr<const EclipseGrid> eg = std::make_shared<const EclipseGrid>(deck , parseContext);
       const std::string* rl_ptr = (rock_list == "no_list") ? 0 : &rock_list;
    -  grid.processEclipseFormat(state->getInputGrid(), periodic_extension, turn_normals, clip_z);
    +  grid.processEclipseFormat(eg, periodic_extension, turn_normals, clip_z);

**Comment by joakim-hove** [comment]
Thanks - I will apply the patch from @pgdr.

**Comment by joakim-hove** [comment]
OK - with some assistance from @pgdr and some rebasing - this should now be Ready&trade;

**Comment by joakim-hove** [comment]
> @atgeirr: Has this reintroduced the need for a complete deck (again)? 

No - already mentioned; but we have certainly created an API - which at least I find easy to use wrong. See discussion here: https://github.com/OPM/opm-parser/issues/795 for a possible solution?

**Pull request merged**